packages/core/src/auto.ts

```diff
@@ -365,6 +366,39 @@ export default class Auto {
       }
     );
 
+    /** 
+     * Determine if repo is behind HEAD of current branch. We do this in
+     * the "afterVersion" hook so the check happens as late as possible.
+     */
+    this.hooks.afterVersion.tapPromise('Check remote for commits', async () => {
+      // Credit from https://github.com/semantic-release/semantic-release/blob/b2b7b57fbd51af3fe25accdd6cd8499beb9005e5/lib/git.js#L179
+      // `true` is the HEAD of the current local branch is the same as the HEAD of the remote branch, falsy otherwise.
+      try {
+        const heads = await execPromise('git', [
+          'ls-remote',
+          '--heads',
+          this.remote,
+          getCurrentBranch()
+        ]);
+        const [, remoteHead] = heads.match(/^(\w+)?/) || [];
+
+        if (remoteHead) {
+          // This will throw if the branch is ahead of the current branch
+          execSync(`git merge-base --is-ancestor ${remoteHead} HEAD`);
```

Comment:
Given #973 I am uncomfortable merging this. If `merge-base` fails on windows this means that `shipit` would never make a release.

@adierkens 🙏